core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java

```diff
@@ -67,102 +82,158 @@ public static void recoverSnapshot(CairoEngine engine) {
-                                    .$(", dst=").$(copyPath)
+                                    .$("could not copy _cv file [src=").$(srcPath)
+                                    .$(", dst=").$(dstPath)
                                     .$(", errno=").$(ff.errno())
                                     .$(']').$();
                         } else {
                             recoveredCVFiles.incrementAndGet();
                             LOG.info()
-                                    .$("recovered _cv file [src=").$(path)
-                                    .$(", dst=").$(copyPath)
+                                    .$("recovered _cv file [src=").$(srcPath)
+                                    .$(", dst=").$(dstPath)
+                                    .$(']').$();
+                        }
+                    }
+
+                    // Go inside SEQ_DIR
+                    srcPath.trimTo(srcPathLen).concat(WalUtils.SEQ_DIR);
+                    srcPathLen = srcPath.length();
+                    srcPath.concat(TableUtils.META_FILE_NAME).$();
+
+                    dstPath.trimTo(dstPathLen).concat(WalUtils.SEQ_DIR);
+                    dstPathLen = dstPath.length();
+                    dstPath.concat(TableUtils.META_FILE_NAME).$();
+
+                    if (ff.exists(srcPath) && ff.exists(dstPath)) {
+                        if (ff.copy(srcPath, dstPath) < 0) {
+                            LOG.error()
+                                    .$("could not copy ").$(TableUtils.META_FILE_NAME).$(" file [src=").$(srcPath)
+                                    .$(", dst=").$(dstPath)
+                                    .$(", errno=").$(ff.errno())
+                                    .$(']').$();
+                        } else {
+                            try {
+                                srcPath.trimTo(srcPathLen);
+                                openSmallFile(ff, srcPath, srcPathLen, memFile, TableUtils.TXN_FILE_NAME, MemoryTag.MMAP_TX_LOG);
+                                long newMaxTxn = memFile.getLong(0L);
+
+                                memFile.smallFile(ff, dstPath, MemoryTag.MMAP_SEQUENCER_METADATA);
+                                long maxStructureVersion = memFile.getLong(SEQ_META_OFFSET_STRUCTURE_VERSION);
+                                long txnMetaMemSize = memFile.getLong(SEQ_META_OFFSET_WAL_LENGTH);
+
+                                if (newMaxTxn >= 0) {
+                                    dstPath.trimTo(dstPathLen);
+                                    openSmallFile(ff, dstPath, dstPathLen, memFile, TXNLOG_FILE_NAME, MemoryTag.MMAP_TX_LOG);
+
+                                    long oldMaxTxn = memFile.getLong(MAX_TXN_OFFSET);
+                                    if (newMaxTxn < oldMaxTxn) {
+                                        memFile.putLong(MAX_TXN_OFFSET, newMaxTxn);
+                                        memFile.putLong(MAX_STRUCTURE_VERSION_OFFSET, maxStructureVersion);
```

Comment:
This must match txn_seq\_meta structure version

Reply:
Here `txn_seq/_txnlog` is patched with values from `txn_seq/_meta`.